src/test/java/org/jabref/gui/externalfiles/FileFilterUtilsTest.java

```diff
@@ -0,0 +1,207 @@
+            LocalDateTime fourthDate = LocalDateTime.of(2021, 1, 2, 0, 0, 0, 0);
+    
+            firstFile.setLastModified(firstDate.atZone(zoneId).toEpochSecond());
+            secondFile.setLastModified(secondDate.atZone(zoneId).toEpochSecond());
+            thirdFile.setLastModified(thirdDate.atZone(zoneId).toEpochSecond());
+            fourthFile.setLastModified(fourthDate.atZone(zoneId).toEpochSecond());
+    
+            // fill the list to be sorted by the tests.
+            files.add(firstPath);
+            files.add(secondPath);
+            files.add(thirdPath);
+            files.add(fourthPath);
+    
+            // fill the expected values lists.
+            expectedSortByDateAscending.add(thirdPath.toString());
+            expectedSortByDateAscending.add(fourthPath.toString());
+            expectedSortByDateAscending.add(secondPath.toString());
+            expectedSortByDateAscending.add(firstPath.toString());
+    
+            expectedSortByDateDescending.add(firstPath.toString());
+            expectedSortByDateDescending.add(secondPath.toString());
+            expectedSortByDateDescending.add(fourthPath.toString());
+            expectedSortByDateDescending.add(thirdPath.toString());
+
+            wrongOrder.add(firstPath.toString());
+            wrongOrder.add(secondPath.toString());
+            wrongOrder.add(thirdPath.toString());
+            wrongOrder.add(fourthPath.toString());
+        }
+    
+        /* Delete the files and clear the path lists. */
+        @AfterEach
+        public void cleanUp() {
+            String dirPath = "src/test/resources/unlinkedFiles";
+
+            File firstFile = new File(dirPath.concat("/firstFile.pdf"));
+            File secondFile = new File(dirPath.concat("/secondFile.pdf"));
+            File thirdFile = new File(dirPath.concat("/thirdFile.pdf"));
+            File fourthFile = new File(dirPath.concat("/fourthFile.pdf"));
+            File directory = new File(dirPath);
+    
+            firstFile.delete();
+            secondFile.delete();
+            thirdFile.delete();
+            fourthFile.delete();
+            directory.delete();
+        }
+    
+        @Test
+        public void sortByDateAscendingPositiveTest() {
+            List<String> actualsList = new ArrayList<String>();
+            List<Path> sortedPaths = fileFilterUtils.sortByDateAscending(files);
+            String[] expecteds = new String[4];
+            String[] actuals = new String[4];
+            for (Path path : sortedPaths) {
+                actualsList.add(path.toString());
+            }
+            actualsList.toArray(actuals);
+            expectedSortByDateAscending.toArray(expecteds);
+            assertArrayEquals(actuals, expecteds);
+        }
+
+        @Test
+        public void sortByDateAscendingNegativeTest() {
+            List<String> actualsList = new ArrayList<String>();
+            List<Path> sortedPaths = fileFilterUtils.sortByDateAscending(files);
```

Comment:
Why not use a stream here? So you don't need to create arrrays
streams().map(Path::toString).collectors(Collection.toList())